### ISSUE TITLE
Not work with 'laravel/tinker' package

### ISSUE THREAD

**Issue by a5867460** [lvht/phpcd.vim#148]
When I create a new project with laravel5.6, phpcd totally not work.
When i remove 'laravel/tinker' package, phpcd works totally fine, but this package were shipped after laravel5.5+ by default, 
Well, I dont need the package so  removed  it was just fine with me, but i think maybe its best to let u know about this situation.
 BTW, neovim version 0.2.3-dev/0.2.2 with deoplete

**Comment by lvht** [comment]
please test the latest code.

**Comment by a5867460** [comment]
works fine now, thanks a lot.
but there's another issue , when i use a component factory's static function which return with a component object like X::db()->,
error happens
log:
[2018-04-19 19:57:31] PHPCD.DEBUG: ReflectionException: Method db does not exist in /home/yy/.nvim/plugged/phpcd.vim/php/PHPCD.php:531 
Stack trace: 
#0 path/to/.nvim/plugged/phpcd.vim/php/PHPCD.php(531): ReflectionClass->getMethod('db') 
#1 path/to/.nvim/plugged/phpcd.vim/php/PHPCD.php(396): PHPCD\PHPCD->typeByReturnType('\\X', 'db', '/home/yy/workdi...') 
#2 [internal function]: PHPCD\PHPCD->functype('\\X', 'db', '/home/yy/workdi...') 
#3 path/to/.nvim/plugged/phpcd.vim/vendor/lvht/msgpack-rpc/src/AbstractServer.php(88): call_user_func_array(Array, Array) 
#4 path/to/.nvim/plugged/phpcd.vim/vendor/lvht/msgpack-rpc/src/AbstractServer.php(65): Lvht\MsgpackRpc\ForkServer->doRequest(Array, Array) 
#5 path/to/.nvim/plugged/phpcd.vim/vendor/lvht/msgpack-rpc/src/AbstractServer.php(44): Lvht\MsgpackRpc\ForkServer->onRequest(Array) 
#6 path/to/.nvim/plugged/phpcd.vim/vendor/lvht/msgpack-rpc/src/ForkServer.php(32): Lvht\MsgpackRpc\ForkServer->onMessage(Array) 
#7 path/to/.nvim/plugged/phpcd.vim/php/main.php(41): Lvht\MsgpackRpc\ForkServer->loop() 
#8 {main} [] []

this happens after alter to the better-reflection.
i can open a new issue for this if its necessory.
phpcd is a greate work, thanks again

**Comment by lvht** [comment]
please off the code php sinister

**Comment by a5867460** [comment]
The exception is fine, cause i can not get any candidates
the situation happens after `0fac411`
```
                $methods = $reflection_class->getPseudoMethods($name);
                $methods = $this->getPseudoMethods($reflection_class, $name);
```

May change the first line to the second? cause seems works for me.

**Comment by lvht** [comment]
@a5867460 please offer you own php source code to trigger error. I will test it.

**Comment by lvht** [comment]
fixed by 94dfea80c287c0b42546a200ffeab54b5d072d7a

**Comment by a5867460** [comment]
works perfect now, thanks a lot
